lib/bundler/errors.rb

```diff
@@ -73,6 +73,27 @@ def message
     status_code(23)
   end
 
+  class TemporaryResourceError < BundlerError
```

Comment:
inherit from `PermissionError` and extract the `action` into a method so it doesn't need to be duplicated?